aws/provider.go

```diff
@@ -195,6 +195,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_ebs_snapshot":                              dataSourceAwsEbsSnapshot(),
 			"aws_ebs_snapshot_ids":                          dataSourceAwsEbsSnapshotIds(),
 			"aws_ebs_volume":                                dataSourceAwsEbsVolume(),
+			"aws_ec2_instance_spot_price":                   dataSourceAwsEc2InstanceSpotPrice(),
```

Comment:
To ensure we align with the [`DescribeSpotPriceHistory` EC2 API naming](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotPriceHistory.html) and future-proof the Terraform naming should this API support spot price history for other resources, let's name this data source either `aws_ec2_spot_price` or `aws_ec2_spot_price_history` (preferably the latter).

```suggestion
			"aws_ec2_spot_price_history":                     dataSourceAwsEc2SpotPriceHistory(),
```

Sorry for the hassle, but this will also require renaming all the functions/files to match.

Reply:
I'm partial to `aws_ec2_spot_price`, because the data source we are introducing is a point-in-time view of the current spot price for an instance; using the term `history` seems misleading, since we are not actually providing a history of the spot price. Happy to make this change if using that term is okay with you @bflad, and I will defer to what you think is best for naming here.

Reply:
`aws_ec2_spot_price` is fine 👍

Reply:
I've updated all references to this name in the PR :+1: